packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,413 @@
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * The increment between successive values when no modifier keys are held
+     * @default 1
+     */
+    stepSize?: number;
+
+    /** The value to display in the input field */
+    value?: number | string;
+
+    /** The callback invoked when `enter` is pressed and when the field loses focus */
+    onConfirm?(value: string): void;
+
+    /** The callback invoked when the value changes */
+    onUpdate?(value: string): void;
+}
+
+export interface INumericStepperState {
+    isInputGroupFocused?: boolean;
+    isButtonGroupFocused?: boolean;
+    shouldSelectAfterUpdate?: boolean;
+    value?: string;
+}
+
+@PureRender
+export class NumericStepper extends AbstractComponent<HTMLInputProps & INumericStepperProps, INumericStepperState> {
+    public static displayName = "Blueprint.NumericStepper";
+
+    public static defaultProps: INumericStepperProps = {
+        buttonPosition: Position.RIGHT,
+        majorStepSize: 10,
+        minorStepSize: 0.1,
+        stepSize: 1,
+        value: NumericStepper.VALUE_EMPTY,
+    };
+
+    private static DECREMENT_KEY = "decrement";
+    private static INCREMENT_KEY = "increment";
+    private static DECREMENT_ICON_NAME = "chevron-down";
+    private static INCREMENT_ICON_NAME = "chevron-up";
+    private static VALUE_EMPTY = "";
+    private static VALUE_ZERO = "0";
+
+    private inputElement: HTMLInputElement;
+
+    public constructor(props?: HTMLInputProps & INumericStepperProps) {
+        super(props);
+        this.state = {
+            shouldSelectAfterUpdate: false,
+            value: this.getValueOrEmptyValue(props),
+        };
+    }
+
+    public render() {
+        const { buttonPosition, className } = this.props;
+
+        const inputGroup = (
+            <InputGroup
+                {...this.removeNonHTMLProps(this.props)}
+                intent={this.props.intent}
+                inputRef={this.inputRef}
+                key="input-group"
+                leftIconName={this.props.leftIconName}
+                onFocus={this.handleInputFocus}
+                onBlur={this.handleInputBlur}
+                onChange={this.handleInputUpdate}
+                onKeyDown={this.handleInputKeyDown}
+                value={this.state.value}
+            />
+        );
+
+        if (buttonPosition === "none" || buttonPosition === null) {
+            // If there are no buttons, then the control group will render the
+            // text field with squared border-radii on the left side, causing it
+            // to look weird. This problem goes away if we simply don't nest within
+            // a control group.
+            return (
+                <div className={className}>
+                    {inputGroup}
+                </div>
+            );
+        } else {
+            // alias this class to avoid line-length lint errors when defining the button group.
```

Comment:
meh, worst case you just disable line length for a moment. or pull the buttons out to variables so you can define them over multiple lines outside of JSX.